backend/src/main/java/de/trailmate/backend/repository/TourRepository.java

```diff
@@ -11,16 +11,28 @@
 import java.util.Map;
 
 
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
 
 @Repository
 public class TourRepository {
 
     private Map<String, Tour> tourList = new HashMap<>();
 
 
+
+
     public List<Tour> getAllTours(){
         return tourList.values().stream().toList();
     }
 
 
+    public Tour getSingleTour(String id) {
+       return tourList.get(id);
+    }
```

Comment:
You could return an Optional so that there is possible error handling if the tour id doesn't exist :)
```suggestion
    public Optional<Tour> getSingleTour(String id) {
       return Optional.ofNullable(tourList.get(id));
    }
```